tests/daemon/unit/api/endpoints/test_logs.py

```diff
@@ -13,50 +16,66 @@
 {"host":"ubuntu","process":"32546","type":"INFO","name":"encode2","uptime":"20210124215151","context":"encode2","workspace_path":"/tmp/jinad/32aa7734-fbb8-4e7a-9f76-46221b512648","log_id":"16ef0bd7-e534-42e7-9076-87a3f585933c","message":"starting jina.peapods.runtimes.zmq.zed.ZEDRuntime..."}
 """
 
-workspace_id = 'acd2fa50-02a6-452d-add7-ff987810c741'
-flow_id = 'f0f90a3d-d2d9-4646-96d5-2560886f6ef0'
-nonexisting_id = 'fff90a3d-d2d9-4646-96d5-2560886ffff0'
+workspace_id = DaemonID('jworkspace')
+flow_id = DaemonID('jflow')
+nonexisting_id = DaemonID('jflow')
 
 
 def _write_to_logfile(content, append=False):
     with open(
-        get_workspace_path(workspace_id, flow_id, 'logging.log'),
+        get_workspace_path(workspace_id, 'logs', flow_id, 'logging.log'),
         'a' if append else 'w+',
     ) as f:
         f.writelines(content)
 
 
-def setup_module():
-    print('setup', get_workspace_path(workspace_id, flow_id))
-    os.makedirs(get_workspace_path(workspace_id, flow_id), exist_ok=True)
-    _write_to_logfile(log_content)
+def _write_to_workspace_logfile(content, append=False):
+    with open(
+        get_workspace_path(workspace_id, 'logs', 'logging.log'),
+        'a' if append else 'w+',
+    ) as f:
+        f.writelines(content)
 
 
-def test_logs_invalid_workspace(fastapi_client):
-    response = fastapi_client.get(f'/logs/{nonexisting_id}/{flow_id}')
-    assert response.status_code == 404
+def _create_flow():
+    flow_store[flow_id] = ContainerItem(
+        metadata=ContainerMetadata(
+            container_id='container_id',
+            container_name='container_name',
+            image_id='image_id',
+            network='',
+            ports={},
+            rest_api_uri='',
+        ),
+        arguments=ContainerArguments(command='command'),
+        workspace_id=workspace_id,
+    )
 
 
-def test_logs_invalid_flow(fastapi_client):
-    response = fastapi_client.get(f'/logs/{workspace_id}/{nonexisting_id}')
-    assert response.status_code == 404
+def setup_module():
+    print('setup', get_workspace_path(workspace_id, flow_id))
+    _create_flow()
```

Comment:
Why is this needed?

Reply:
the new log endpoint is looking up the workspace id from the entity (pod/pea/flow) id. So if it does not exist it will fail.

so it does basically
`correct_store[log_id].workspace_id`